### ISSUE TITLE
Gradual unfreezing the layers during training. 

### ISSUE THREAD

**Issue by senhorinfinito** [ultralytics/yolov5#9677]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

In [ULMFIT](https://arxiv.org/abs/1801.06146) three things they have proposed to the fine-tune the model
1. Slanted triangular learning rate.
2. gradual unfreezing
3. discriminative fine tuning.

I wanted to implement only gradual unfreezing.

### Use case

Unfreezing the layers in the middle of training, a specific epoch. 

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@senhorinfinito see Transfer Learning with Frozen Layers tutorial below.

## Tutorials

- [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)  🚀 RECOMMENDED
- [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)  ☘️ RECOMMENDED
- [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
- [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) 🌟 NEW
- [TFLite, ONNX, CoreML, TensorRT Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
- [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
- [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
- [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
- [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
- [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)
- [Architecture Summary](https://docs.ultralytics.com/yolov5/tutorials/architecture_description) 🌟 NEW
- [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)
- [Roboflow for Datasets, Labeling, and Active Learning](https://docs.ultralytics.com/yolov5/tutorials/roboflow_datasets_integration)  🌟 NEW
- [ClearML Logging](https://docs.ultralytics.com/yolov5/tutorials/clearml_logging_integration) 🌟 NEW
- [Deci Platform](https://github.com/ultralytics/yolov5/wiki/Deci-Platform) 🌟 NEW
- [Comet Logging](https://docs.ultralytics.com/yolov5/tutorials/comet_logging_integration) 🌟 NEW

Good luck 🍀 and let us know if you have any other questions!

**Comment by senhorinfinito** [comment]
@glenn-jocher  Thanks for response. I read all from that article (issue) & have created a PR to implement unfreezing the layers. Please review & suggest if any changes are need to done.
